views/AppView.js

```diff
@@ -0,0 +1,55 @@
+var AppView = Backbone.View.extend({
+  el: $('body'),
+
+  events: {
+    'click .search-video': 'handleSearchClick',
+    'click .title-link' : 'handleLinkClick'
+  },
+
+  handleSearchClick: function () {
+    var searchTermInput = this.$('.search-form').val();
+    this.model.set('searchTerm', searchTermInput);
+  },
+
+  handleLinkClick: function (e) {
+    var clickedVideoId = $(e.currentTarget).data().id;
+    this.model.updateMainVideo(clickedVideoId)
+  },
+
+  initialize: function () {
+    this.listenTo(this.model.get('videos'), 'reset', this.renderPage);
+    this.listenTo(this.model, 'change:searchTerm', this.fetchOnSearch);
+    this.listenTo(this.model, 'change:mainVideo', this.renderNewMainVideo);
+  },
+
+  fetchOnSearch: function () {
```

Comment:
Again, could be moved to the model (or collection) as the view doesn't need to be concerned with how data is fetched.